databricks/koalas/indexing.py

```diff
@@ -75,12 +75,169 @@ def _unfold(key, col):
     return rows_sel, cols_sel
 
 
-class SparkDataFrameLocator(object):
+class LocIndexer(object):
     """
-    A locator to slice a group of rows and columns by conditional and label(s).
+    Access a group of rows and columns by label(s) or a boolean Series.
 
-    Allowed inputs are a slice with all indices or conditional for rows, and string(s) or
-    :class:`Column`(s) for cols.
+    ``.loc[]`` is primarily label based, but may also be used with a
+    conditional boolean Series derived from the DataFrame or Series.
+
+    Allowed inputs are:
+
+    - A single label, e.g. ``5`` or ``'a'``, (note that ``5`` is
+      interpreted as a *label* of the index, and **never** as an
+      integer position along the index) for column selection.
+
+      .. warning:: A single label for row selection is not allowed
+
+    - A list or array of labels, e.g. ``['a', 'b', 'c']``.
+
+      .. warning:: With a list or array of labels for row selection,
+          Koalas does not preserve the given order but preserves the data order.
+
+    - A slice object with labels, e.g. ``'a':'f'``.
+
+      .. warning:: Note that contrary to usual python slices, **both** the
+          start and the stop are included
+
+      .. warning:: The step of the slice is not allowed
+
+      .. warning:: With a slice, Koalas works as a filter between the range
+
+    - A conditional boolean Series derived from the DataFrame or Series
+
+    Not allowed inputs which pandas allows are:
+
+    - A boolean array of the same length as the axis being sliced,
+      e.g. ``[True, False, True]``.
+    - A ``callable`` function with one argument (the calling Series, DataFrame
+      or Panel) and that returns valid output for indexing (one of the above)
+
+    .. warning:: MultiIndex is not supported yet.
+
+    See Also
+    --------
+    Series.loc : Access group of values using labels.
+
+    Examples
+    --------
+    **Getting values**
+
+    >>> df = ks.DataFrame([[1, 2], [4, 5], [7, 8]],
+    ...                   index=['cobra', 'viper', 'sidewinder'],
+    ...                   columns=['max_speed', 'shield'])
+    >>> df
+                max_speed  shield
+    cobra               1       2
+    viper               4       5
+    sidewinder          7       8
+
+    A single label for row selection is not allowed.
+
+    >>> df.loc['viper']
```

Comment:
is this just a filter? maybe we can implement this in the future as a todo.

Reply:
It is not just a filter. pandas returns `pd.Series` for that case:

```py
>>> import pandas as pd
>>> pdf = pd.DataFrame([[1, 2], [4, 5], [7, 8]], index=['cobra', 'viper', 'sidewinder'], columns=['max_speed', 'shield'])
>>> pdf
            max_speed  shield
cobra               1       2
viper               4       5
sidewinder          7       8
>>> pdf.loc['viper']
max_speed    4
shield       5
Name: viper, dtype: int64
```

whereas if the index is duplicated, then returns `pd.DataFrame`:

```py
>>> pdf = pd.DataFrame([[1, 2], [4, 5], [7, 8], [9, 10]], index=['cobra', 'viper', 'sidewinder', 'viper'], columns=['max_speed', 'shield'])
>>> pdf.loc['viper']
       max_speed  shield
viper          4       5
viper          9      10
```

Maybe we should just return `DataFrame`?

Reply:
Is it not possible to follow pandas? 

btw this is future work anyway. maybe create a ticket so we can discuss there.

Reply:
Here's the issue corresponding this (#383).